src/compiler/llvm/mir.rs

```diff
@@ -418,6 +578,29 @@ impl<'p, 'module, 'ctx> FunctionBuilder<PointerValue<'ctx>, BasicValueEnum<'ctx>
         self.program.builder.build_store(l, v);
     }
 
+    fn store(&mut self, span: Span, l: &LValue, r: BasicValueEnum<'ctx>) {
```

Comment:
I think the old `assign` method can be deleted.

Reply:
Use this store, or update the trait to have a `store` method for each variant of `LValue`?  That would account for providing the context needed for things like, properly mapping `ReturnPointer` to the right location in LLVM while also removing the need for the Transformer to understand the structure of the various enumerations in MIR.

Reply:
Will leave for now, may change in teh future.